SimCore/src/SimCore/PhotoNuclearModels/BertiniEventTopologyProcess.cxx

```diff
@@ -5,9 +5,14 @@ void BertiniEventTopologyProcess::cleanupSecondaries() {
   int secondaries{theParticleChange.GetNumberOfSecondaries()};
   // Geant4 won't clean up this memory for us by default
   for (int i{0}; i < secondaries; ++i) {
+    // disable clang's use-after-free warning for this delete
+    // as far as we (@tomeichlersmith and @EinarElen) can tell,
+    // this is correct and clang is just getting confused since
+    // Geant4's inlined delete method is written in a wacky way
+#ifndef __clang_analyzer__
     auto secondary{theParticleChange.GetSecondary(i)->GetParticle()};
-    pDynamicParticleAllocator->FreeSingle(secondary);
-    pDynamicParticleAllocator->destroy(secondary);
+    delete secondary;
```

Comment:
why do we create `secondary` just so we can delete it in the next row? :D (sorry for my ignorance)

Reply:
Its not creating, we are getting the _pointer_ and then deleting the _object_.

This function is called when we are disposing of an event that is not accepted by the filter and so we need to cleanup the particles that were created by the hadronic process. Usually, the particles created by the hadronic process would continue propagating through the rest of the sim like other particles created by other processes.